tests/performance/instrumentation_step_22.cc

```diff
@@ -0,0 +1,677 @@
+    tmp *= -1;
+    tmp += system_rhs.block(0);
+
+    A_inverse.vmult(solution.block(0), tmp);
+
+    constraints.distribute(solution);
+  }
+}
+
+
+template <int dim>
+void
+StokesProblem<dim>::output_results(
+  const unsigned int /*refinement_cycle*/) const
+{
+  std::vector<std::string> solution_names(dim, "velocity");
+  solution_names.emplace_back("pressure");
+
+  std::vector<DataComponentInterpretation::DataComponentInterpretation>
+    data_component_interpretation(
+      dim, DataComponentInterpretation::component_is_part_of_vector);
+  data_component_interpretation.push_back(
+    DataComponentInterpretation::component_is_scalar);
+
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution,
+                           solution_names,
+                           DataOut<dim>::type_dof_data,
+                           data_component_interpretation);
+  data_out.build_patches();
+}
+
+
+template <int dim>
+void
+StokesProblem<dim>::refine_mesh()
+{
+  Vector<float> estimated_error_per_cell(triangulation.n_active_cells());
+
+  const FEValuesExtractors::Scalar pressure(dim);
+  KellyErrorEstimator<dim>::estimate(
+    dof_handler,
+    QGauss<dim - 1>(degree + 1),
+    std::map<types::boundary_id, const Function<dim> *>(),
+    solution,
+    estimated_error_per_cell,
+    fe.component_mask(pressure));
+
+  GridRefinement::refine_and_coarsen_fixed_number(triangulation,
+                                                  estimated_error_per_cell,
+                                                  0.3,
+                                                  0.0);
+  triangulation.execute_coarsening_and_refinement();
+}
+
+
+template <int dim>
+Measurement
+StokesProblem<dim>::run()
+{
+  std::map<std::string, unsigned long long> cycle_count;
+
+  {
+    std::vector<unsigned int> subdivisions(dim, 1);
+    subdivisions[0] = 4;
+
+    const Point<dim> bottom_left = (dim == 2 ?                //
+                                      Point<dim>(-2, -1) :    // 2d case
+                                      Point<dim>(-2, 0, -1)); // 3d case
+
+    const Point<dim> top_right = (dim == 2 ?              //
+                                    Point<dim>(2, 0) :    // 2d case
+                                    Point<dim>(2, 1, 0)); // 3d case
+
+    GridGenerator::subdivided_hyper_rectangle(triangulation,
+                                              subdivisions,
+                                              bottom_left,
+                                              top_right);
+  }
+
+  for (const auto &cell : triangulation.active_cell_iterators())
+    for (const auto &face : cell->face_iterators())
+      if (face->center()[dim - 1] == 0)
+        face->set_all_boundary_ids(1);
+
+
+  switch (get_testing_environment())
+    {
+      case TestingEnvironment::light:
+        triangulation.refine_global(4 - dim);
+        break;
+      case TestingEnvironment::medium:
+        /* fallthrough */
```

Comment:
```suggestion
#ifdef DEAL_II_HAVE_CXX17
        [[fallthrough]];
#endif
```
See https://en.cppreference.com/w/cpp/language/attributes/fallthrough

Reply:
We also have a deal.II version, see https://github.com/dealii/dealii/blob/951db6b4eda1781f9ae1b983df2ffe9162bea546/include/deal.II/base/config.h.in#L175 but you really don't get a compiler warning for empty cases.

Reply:
I have changed switched the comment to our `DEAL_II_FALLTHROUGH` macro.